### PR TITLE
dapp: fix correct match of build modes and paths

### LINKED ISSUE #2269: Staging is not working
## Steps to Reproduce
1. Go to https://lightclient.raiden.network/staging

## Expected Result
- I see the dApp

## Actual Result
- I don't see the dApp
grafik

## Additional Infos (Optional) 
```
Tested with latest
- Firefox
- Edge
```

### PR DESCRIPTION
**Pull request by weilbith**
Fixes #2269  

**Short description**
The commit message is huuuuge (sorry for that) and should basically explain everything. Checkout the below steps how to do a basic check of this fix gglocally.

**Definition of Done**

- [x] Steps to manually test the change have been documented
- [x] Acceptance criteria are met
- [ ] Change has been manually tested by the reviewer (dApp) 

**Steps to manually test the change (dApp)**

1. Create somewhere a new directory which will later be used as root to serve static assets from
2. Build the dApp in production mode `pnpm run build --filter raiden-dapp -- --mode production` (lets be explicit with the mode)
3. Copy the build to the created directory `cp -r raiden-dapp/dist/* /to/the/static/assets/dir`
4. Build the dApp in staging mode `pnpm run build --filter raiden-dapp -- --mode staging`
5. Copy the build into the `/staging` sub-folder of the assets directory.
6. Serve the static assets locally (e.g. `npx serve /to/the/static/assets/dir`)
7. Visit the dApp in the browser on `/` and `/staging`, make sure that both work but both do not allow mainnet connecitions

### CODE REVIEW

File: raiden-dapp/vue.config.js [hunk 1]
```diff
@@ -5,12 +5,9 @@ module.exports = {
...
   parallel: !process.env.CIRCLECI,
-  publicPath:
-    process.env.NODE_ENV !== 'production'
-      ? './'
-      : process.env.DEPLOYMENT === 'staging'
-      ? '/staging/'
-      : '/',
+  publicPath: process.env.VUE_APP_PUBLIC_PATH
+    ? process.env.VUE_APP_PUBLIC_PATH
+    : '/',
```
Comment (andrevmatos) [inline raiden-dapp/vue.config.js:10]: We need to fallback to relative paths (`./`) if on development mode, or else this breaks the serve_pr script

Comment (weilbith) [reply raiden-dapp/vue.config.js:10]: Ah thats what it is for. Wait a seconds I'll find a solution.

Comment (weilbith) [reply raiden-dapp/vue.config.js:10]: Here we go. I'll test it using the serve pr script as soon as the latest CI run through.

**Comment by weilbith** [comment]
Damn it. I think due to the parametrization the server script does not work anymore.

**Comment by weilbith** [comment]
Fixed. Don't hesitate to give a try. :upside_down_face:

**Review by andrevmatos — APPROVED** [review]
lgtm, tested here with both script and docker image. let's see it deployed on staging to be sure. thanks

**Comment by weilbith** [comment]
It works :wink:

**Pull request merged**